game/scripts/vscripts/units/shielder.lua

```diff
@@ -0,0 +1,212 @@
+
+--[[ Taken from bb template
+if ShielderAI == nil then
+  DebugPrint ( 'creating new BossAI object' )
+  ShielderAI = class({})
+
+  Debug.EnabledModules['boss:*'] = false
+end]]
+
+LinkLuaModifier( "modifier_boss_phase_controller", "modifiers/modifier_boss_phase_controller", LUA_MODIFIER_MOTION_NONE )
+
+function Spawn (entityKeyValues)
+  thisEntity:FindAbilityByName("boss_shielder_shield")
+
+  thisEntity:SetContextThink( "ShielderThink", partial(ShielderThink, thisEntity) , 1)
+  print("Starting AI for " .. thisEntity:GetUnitName() .. " " .. thisEntity:GetEntityIndex())
+
+  ABILITY_shield = thisEntity:FindAbilityByName("boss_shielder_shield")
+
+  local phaseController = thisEntity:AddNewModifier(thisEntity, ABILITY_shield, "modifier_boss_phase_controller", {})
+  phaseController:SetPhases({ 66, 33 })
+  phaseController:SetAbilities({
+    "boss_shielder_shield"
+  })
+end
+
+--[[
+ShielderAI.IDLE = 1
+ShielderAI.AGRO = 2
+ShielderAI.LEASHING = 3
+ShielderAI.DEAD = 4
+]]
+
+function Create (unit, options)
+  options = {}
+  options.tier = 1
+
+  local state = {
+    handle = unit,
+    origin = unit:GetAbsOrigin(),
+    leash = 1500,
+    agroDamage = 100,
+    tier = 1,
+    currentDamage = 0,
+    state = 1,
+    customAgro = true,
+
+    deathEvent = Event()
+  }
+
+  unit:OnHurt(function (keys)
```

Comment:
this can be simpler written as `unit:OnHurt(partial(HurtHandler, state))` hooray for functional goodness!

Also I'm surprised these methods exist at create time. Cool!

That said, the hurt handler does nothing so you can just do the death handler...